### ISSUE TITLE
Centering on the same hunk position

### ISSUE THREAD

**Issue by wavexx** [pidu/git-timemachine#29]
I used briefly `git-timemachine` to browse though the history of some files as an alternative to looking at complex diffs and I really appreciate the effort.

I noticed though that it's actually hard to follow what's going since the cursor is kept on the same line position irregardless if the file has new or deleted lines. It works nicely for tiny files, but browsing quickly becomes useless at more than 50% of the buffer and/or for large files.

To make timemachine instantly more useful it would help to try to bring the cursor to the same hunk position across each revision (if it still exists). I guess a robust way would be to sum line-count changes for all the hunks before the current cursor position and move the cursor according to that. The downside is that it requires parsing the diff of each revision you're moving to.

**Comment by pidu** [comment]
Sounds like a good idea, try it out and create a PR :-)

**Comment by wavexx** [comment]
I suspected such an answer ;). I might try if I find some time, but definitely not soon.
Can you think of a better way / and or an alternative approach of doing diff + show?

I was spoiled by this approach by ``p4v`` (the perforce GUI), and it makes a huge difference.

**Comment by pidu** [comment]
Seems like a tricky problem to solve, I'm not well versed enough with git on the command line to come up with a better suggestion right now. I agree that a well working implementation would really help usability but I'm not seeing myself attempting this right now.
